src/Compilers/CSharp/Test/Semantic/Semantics/MethodTypeInferenceTests.cs

```diff
@@ -633,6 +635,251 @@ static void Main(string[] args)
+            foreach (var discardOut in GetDiscardIdentifiers(tree))
+            {
+                checkDiscard(model, discardOut, "System.String!");
+            }
+
+            // out var _, out T _
+            foreach (var discardDecl in GetDiscardDesignations(tree))
+            {
+                checkDiscard(model, discardDecl, "System.String!");
+            }
+        }
+
+        [Fact]
+        public void Bug50782_3()
+        {
+            string source = """
+#nullable enable
+
+interface IOperation<T> 
+{ }
+
+public class StringOperation : IOperation<string> 
+{ }
+
+public class C 
+{   
+    static void Main() 
+    {
+        TestA(new StringOperation(), out string? discardA);
+        TestA(new StringOperation(), out string? _);
+
+        TestC(out string? discardC);
+        TestC(out string? _);
+    }
+   
+    static void TestA<T>(IOperation<T> operation, out T result) => result = default;
+    static void TestC(out string result) => result = "";
+}
+""";
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyDiagnostics(new[] {                    
+                // (20,77): warning CS8601: Possible null reference assignment.
+                //     static void TestA<T>(IOperation<T> operation, out T result) => result = default;
+                Diagnostic(ErrorCode.WRN_NullReferenceAssignment, "default").WithLocation(20, 77)
+            });
+
+            var tree = comp.SyntaxTrees.Single();
+            var model = comp.GetSemanticModel(tree);
+
+            // out _
+            foreach (var discardOut in GetDiscardIdentifiers(tree))
+            {
+                checkDiscard(model, discardOut, "System.String?");
+            }
+
+            // out var _, out T _
+            foreach (var discardDecl in GetDiscardDesignations(tree))
+            {
+                checkDiscard(model, discardDecl, "System.String?");
+            }
+        }
+
+        [Fact]
+        public void Bug50782_4()
+        {
+            string source = """
+#nullable enable
+void M((string, string?) tuple)
+{
+    (_, _) = tuple;
+    (string _, string _) = tuple;
+}
+""";
+
+            var comp = CreateCompilation(source, targetFramework: TargetFramework.Net70);
+            comp.VerifyDiagnostics(new[] { 
+                 // (2,6): warning CS8321: The local function 'M' is declared but never used
+                // void M((string, string?) tuple)
+                Diagnostic(ErrorCode.WRN_UnreferencedLocalFunction, "M").WithArguments("M").WithLocation(2, 6),
+                // (5,28): warning CS8600: Converting null literal or possible null value to non-nullable type.
+                //     (string _, string _) = tuple;
+                Diagnostic(ErrorCode.WRN_ConvertingNullableToNonNullable, "tuple").WithLocation(5, 28)
+            });
+        }
+
+        private static void checkDiscard(SemanticModel model, DiscardDesignationSyntax discard, string type) {
```

Comment:
Consider renaming these to `CheckDiscard` to match conventions elsewhere in the project.